src/main/scala/interfaces/Verification.scala

```diff
@@ -7,13 +7,52 @@
 package viper.silicon.interfaces
 
 import viper.silicon.interfaces.state.Chunk
+import viper.silicon.state.terms.Term
 import viper.silicon.state.{State, Store}
 import viper.silver.verifier.{Counterexample, Model, VerificationError}
-import viper.silicon.state.terms.Term
 
 /*
  * Results
  */
+class VerificationResultWrapper(verificationResultsarg: Seq[VerificationResult]) {
+  def verificationResults: Seq[VerificationResult] = verificationResultsarg
+
+  def &&(other: => VerificationResultWrapper): VerificationResultWrapper = {
+    //Filter out multiple Successes
+    var combinedSeq = this.verificationResults ++ other.verificationResults
+    val hasSuccess = combinedSeq.contains(Success())
+    combinedSeq = combinedSeq flatMap {
+      case _: Success => None
+      case vr: VerificationResult => Some(vr)
+    }
+    if (combinedSeq.isEmpty && hasSuccess) combinedSeq = Seq(Success())
+    new VerificationResultWrapper(combinedSeq)
+  }
+  // short circuit "and"
+  def &&&(other: => VerificationResultWrapper): VerificationResultWrapper = { //TODO:J check if this is really needed
```

Comment:
While method/operator `&&` conveys some intuition, it will be difficult to make an educated guess about the difference between `&&` and `&&&`. Is `&&&` really necessary as an operator? Can it be given a more self-explaining name?